miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java

```diff
@@ -252,8 +290,293 @@ private static BoxableId parseEntityIdentifier(String identifier) {
     }
   }
 
-  public void setBoxService(BoxService boxService) {
-    this.boxService = boxService;
+  public static class ScannerPreparationRequest {
+    private String scannerName;
+    private Integer rows;
+    private Integer columns;
+
+    public String getScannerName() {
+      return scannerName;
+    }
+
+    public void setScannerName(String scannerName) {
+      this.scannerName = scannerName;
+    }
+
+    public Integer getRows() {
+      return rows;
+    }
+
+    public void setRows(Integer rows) {
+      this.rows = rows;
+    }
+
+    public Integer getColumns() {
+      return columns;
+    }
+
+    public void setColumns(Integer columns) {
+      this.columns = columns;
+    }
+  }
+
+  /**
+   * Prepares the associated Box Scanner for scanning.
+   * 
+   * @param requestData
+   */
+  @RequestMapping(value = "/rest/box/prepare-scanner", method = RequestMethod.POST)
+  public void prepareBoxScanner(@RequestBody(required = true) ScannerPreparationRequest requestData) {
+    if (requestData.getRows() == null || requestData.getColumns() == null) {
+      throw new RestException("Box size not specified", Status.BAD_REQUEST);
+    } else if (isStringEmptyOrNull(requestData.getScannerName())) {
+      throw new RestException("No scanner specified", Status.BAD_REQUEST);
+    }
+    BoxScanner boxScanner = boxScanners.get(requestData.getScannerName());
+    if (boxScanner == null) {
+      throw new RestException("Invalid scanner specified", Status.BAD_REQUEST);
+    }
+    try {
+      boxScanner.prepareScan(requestData.getRows(), requestData.getColumns());
+    } catch (IntegrationException e) {
+      throw new RestException("Could not find the scanner", Status.INTERNAL_SERVER_ERROR);
+    } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Combine this catch with the one at line 341, which has the same body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2147)